modules/ximgproc/src/weighted_median_filter.cpp

```diff
@@ -55,51 +55,64 @@ using namespace cv::ximgproc;
 void from32FTo32S(Mat &img, Mat &outImg, int nI, float *mapping)
 {
     int rows = img.rows, cols = img.cols;
-    int alls = rows * cols;
+    size_t alls = (size_t)rows * cols;
 
+    CV_Assert(img.isContinuous());
     float *imgPtr = img.ptr<float>();
     typedef pair<float,int> pairFI;
-    pairFI *data = (pairFI *)malloc(alls*sizeof(pairFI));
+    std::vector<pairFI> data(alls);
 
     // Sort all pixels of the image by ascending order of pixel value
-    for(int i=0;i<alls;i++){
-        data[i].second = i;
-        data[i].first = imgPtr[i];
+    for (size_t i = 0; i < alls; i++)
+    {
+        pairFI& d = data[i];
+        d.second = i;
+        d.first = imgPtr[i];
     }
-    sort(data,data+alls);
+
+    struct PixelValueOrder {
+        static bool compare(const pairFI &a, const pairFI &b) {
+            return a.first < b.first;
+        }
+    };
+    sort(data.begin(), data.end(), PixelValueOrder::compare);
 
     // Find lower bound and upper bound of the pixel values
-    double maxVal,minVal;
-    minMaxLoc(img,&minVal,&maxVal);
-    float maxRange = (float)(maxVal - minVal);
-    float th = 1e-5f;
+    double maxVal = data[alls - 1].first, minVal = data[0].first;
+
+    const float maxRange = (float)(maxVal - minVal);
 
     float l = 0, r = maxRange*2.0f/nI;
     // Perform binary search on error bound
-    while(r-l > th)
+    while (r > l)
     {
-        float m = (r+l)*0.5f;
+        float m = (r + l) * 0.5f;
+        if (m == r || m == r)
```

Comment:
is that a typo here ?

also, `m`  is set to either `l` or `r` below, based on the result in `suc`, so it it meant:

    if (m==r || m==l)

it will bail out in the 2nd iteration already.